tests/unit/samplegen/test_samplegen.py

```diff
@@ -127,30 +162,30 @@ def test_loop_collection_extra_kword():
                      "body": [{"print":
                                ["Mollusc of class: %s", "m.class"]}]}}
     with pytest.raises(samplegen.BadLoop):
-        samplegen.ResponseValidator([loop])
+        samplegen.Validator().validate_response([loop])
 
 
 def test_loop_collection_missing_kword():
     loop = {"loop": {"collection": "$resp.molluscs",
                      "body": [{"print":
                                ["Mollusc of class: %s", "m.class"]}]}}
     with pytest.raises(samplegen.BadLoop):
-        samplegen.ResponseValidator([loop])
+        samplegen.Validator().validate_response([loop])
 
 
 def test_loop_collection_missing_kword2():
     loop = {"loop": {"collection": "$resp.molluscs",
                      "body": [{"print":
                                ["Mollusc: %s", "m.class"]}]}}
     with pytest.raises(samplegen.BadLoop):
-        samplegen.ResponseValidator([loop])
+        samplegen.Validator().validate_response([loop])
```

Comment:
what's the difference with the previous case?

Reply:
I reworked Validator to save the names of the input parameters when `validate_and_transform_request` gets called; the validation then happens in explicit methods.

The alternatives were to either pass in an output parameter lexical scope to the request validation function (gross) or change the return signature to return a 2-tuple of transformed request and input parameter set (also gross).

Now that there's enough state to justify an object, I think that having a sample validator object simplifies a few things and makes subsequent changes more tractable.

Reply:
Sorry, I meant the difference with the previous test case: `test_loop_collection_missing_kword` and `test_loop_collection_missing_kword2`

Reply:
Can't remember; I'll yank it in the next diff.